ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java

```diff
@@ -461,24 +470,28 @@ private void enforceRemoteConnectionLimits() {
         .forEach(
             conn -> {
               LOG.debug(
-                  "Too many remotely initiated connections. Disconnect low-priority connection: {}",
-                  conn);
+                  "Too many remotely initiated connections. Disconnect low-priority connection: {}, maxRemote={}",
+                  conn,
+                  maxRemotelyInitiatedConnections);
               conn.disconnect(DisconnectReason.TOO_MANY_PEERS);
             });
   }
 
   private void enforceConnectionLimits() {
-    if (connectionsById.size() < maxConnections) {
+    if (connectionsById.size() < upperBoundConnections) {
       // Nothing to do - we're under our limits
       return;
     }
 
     getActivePrioritizedConnections()
-        .skip(maxConnections)
+        .skip(upperBoundConnections)
```

Comment:
hmmm maybe?

Reply:
I think it's correct as is. the getActivePrioritizedConnections() sorts so that preferred connections come first. Then we skip maxPeers, and from what's remaining, we only want to consider for disconnection the ones that match the filter. If we filter first, then we would keep maxPeers number of ones that match the filter rather than maxPeers of total number.

Reply:
sounds good